Calculator_exercise.py

```diff
@@ -0,0 +1,299 @@
+
+    Returns: str - The return from the multiply function.
+    """
+
+    matches = re.findall(r'-?\d+/-?\d+', peace)
+    if len(matches) > 0:
+        for match in matches:
+            matches2 = re.findall(r'-?\d+', match)
+            if matches2[1] != '0':
+                peace = peace.replace(match, str(
+                    int(matches2[0]) / int(matches2[1])))
+
+            else:
+                print("Cannot divide by zero!")
+                main()
+
+        return divide(peace)
+
+    else:  # No more '/' -
+        return multiply(peace)
+
+
+def power(peace):
+    """solve the '^' in the expression and send it to divide.
+
+    Args:
+        peace (str): The expression from 'modulo'.
+
+    Returns: str - The return from the divide function.
+    """
+
+    matches = re.findall(r'-?\d+\^-?\d+', peace)
+    if len(matches) > 0:
+        for match in matches:
+            matches2 = re.findall(r'-?\d+', match)
+            peace = peace.replace(match,
+                                  str(int(matches2[0]) ** int(matches2[1])))
+        return power(peace)
+
+    else:  # No more '^' -
+        return divide(peace)
+
+
+def modulo(peace):
+    """Find the remainder of num/num2 and send the expression to power.
+
+    Args:
+        peace (str): The expression from 'minimum'.
+
+    Returns: str - The return from the power function.
+    """
+
+    matches = re.findall(r'-?\d+%-?\d+', peace)
+    if len(matches) > 0:
+        for match in matches:
+            matches2 = re.findall(r'-?\d+', match)
+            peace = peace.replace(match,
+                                  str(int(matches2[0]) % int(matches2[1])))
+        return modulo(peace)
+
+    else:  # No modulo left -
+        return power(peace)
+
+
+def minimum(peace):
+    """Find the smaller number of the two and send the expression to modulo.
+
+    Args:
+        peace (str): The expression from 'maximum'.
+
+    Returns: str - The return from the modulo function.
+    """
+
+    matches = re.findall(r'-?\d+&-?\d+', peace)
+    if len(matches) > 0:
+        for match in matches:
+            matches2 = re.findall(r'-?\d+',  match)
+            int_matches = [int(x) for x in matches2]
+            peace = peace.replace(match, str(min(int_matches)))
+        return minimum(peace)
+
+    else:  # No more '&' -
+        return modulo(peace)
+
+
+def maximum(peace):
+    """Find the larger number between the two and send expression to minimum.
+
+    Args:
+        peace (str): The expression from 'average'.
+
+    Returns: str - The return from the minimum function.
+    """
+
+    matches = re.findall(r'-?\d+\$-?\d+', peace)
+    if len(matches) > 0:
+        for match in matches:
+            matches2 = re.findall(r'-?\d+', match)
+            int_matches = [int(x) for x in matches2]
+            peace = peace.replace(match, str(max(int_matches)))
+        return maximum(peace)
+
+    else:  # No more '$' -
+        return minimum(peace)
+
+
+def average(peace):
```

Comment:
Average should not get precedence over max or min. Whichever comes first (is more to the left) in the equation should be performed first. As you wrote it you will have a problem with 

3 $ 1 @ 3

 If you take the average before the max you will get different (incorrect) results.

You have the same issue with the other equal priority operations.

Note: This will take some major rewriting to fix. Do yourself a favor and write out the logic of how you plan to do this before you start coding it. There is a simple and clean way to handle this and an ugly and messy way.